Sources/Core/CompilerInfo.swift

```diff
@@ -0,0 +1,69 @@
+/// Describes the compiler; used as source of truth for conditional compilation.
+public struct CompilerInfo {
+
+  /// A version number with multiple parts; form: x(.y)*
+  public struct VersionNumber: Codable, Comparable {
+
+    public let parts: [Int]
+
+    public init(_ p: [Int]) {
+      parts = p
+    }
+
+    public static func < (lhs: Self, rhs: Self) -> Bool {
+      return lhs.parts.lexicographicallyPrecedes(rhs.parts)
+    }
+
+  }
+
+  /// The operating-system used when performing the compilation.
+  let os: String
+  /// The architecture of the machine doing the compilation.
```

Comment:
```suggestion

    /// The architecture of the machine doing the compilation.
```
We add newlines between every property declaration.